### PR TITLE
[WFCORE-2953] Add documentation for the socket-handler on the logging subsystem.

### PR DESCRIPTION
**Pull request by jamezp**
This is documentation for [WFCORE-2953](https://issues.jboss.org/browse/WFCORE-2953) [WildFly Core PR](https://github.com/wildfly/wildfly-core/pull/3386) and the [proposal PR](https://github.com/wildfly/wildfly-proposals/pull/103).

The first commit just cleans up the handler documentation a bit as well as remove the common_attributes document in favor of using the maven plugin configuration.

### CODE REVIEW

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Logging_Handlers.adoc [hunk 1]
```diff
@@ -1,446 +1,126 @@
...
-* <<autoflush,#autoflush>>
-* <<enabled,#enabled>>
-* <<encoding,#encoding>>
-* <<filter-spec,#filter-spec>>
-* <<formatter,#formatter>>
-* <<level,#level>>
-* <<named-formatter,#named-formatter>>
-* <<target,#target>>
+A `custom-handler` allows you to define any {oracle-javadoc}/java/util/logging/Handler.html[handler] as a handler that
+can be assigned to a logger or aa <<async-handler,`async-handler`>>.
```
Comment (nziakova) [inline docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Logging_Handlers.adoc]: Probably a typo "aa async-handler" -> "an async-handler" ?

File: docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Logging_Handlers.adoc [hunk 2]
```diff
@@ -1,446 +1,126 @@
...
+A `periodic-size-rotating-file-handler` is a handler that writes log messages to the specified file. The file rotates on
+the date pattern specified in the `suffix` attribute or the `rotate-size` attribute. The suffix must be a valid
+pattern recognized by the {oracle-javadoc}/java/text/SimpleDateFormat.html[`java.text.SimpleDateFormat`] and must
+not rotate on seconds or milliseconds.
 
-[[attributes-3]]
-==== Attributes
+The `max-backup-index` works differently on this handler than the
+<<size-rotating-file-handler,`size-rotating-file-handler`>>. The date suffix of the file to be rotated must be the
+same as the current expected suffix. For example with a suffix patter of `yyyy-MM` and a `rotate-size` of `10m` the
```
Comment (nziakova) [inline docs/src/main/asciidoc/_admin-guide/subsystem-configuration/Logging_Handlers.adoc]: Typo: "suffix patter" -> "suffix pattern"

**Comment by nziakova** [comment]
The docs does not mention SSL/TLS TCP which is listed in the analysis doc as requirement. Maybe the docs could show an example of usage similar to the UDP one, WDYT?

**Comment by jamezp** [comment]
@nziakova For SSL/TLS the idea was the `socket-binding` configuration should cover that. I did make a link to the docs. Would that be enough or do you think we need an example there as well? My only simple objection is we'd probably want to do both an Elytron and legacy example. I just don't want that part of the docs going stale.

**Comment by nziakova** [comment]
@jamezp The link should be enough, but there's no SSL/TLS example at Admin_Guide#socket-binding-groups. Maybe we only need the Elytron example for community docs?

**Comment by nziakova** [comment]
@jamezp I cannot find any SSL setting on the socket-binding, how do you configure it?

**Comment by jamezp** [comment]
@nziakova You're correct. I'll have to rework that as I misunderstood how it works. Sorry about the confusion.

**Comment by nziakova** [comment]
@jamezp Ah, ok. I wasn't sure if I didn't miss anything. So should the SSL config be part of the logging subsystem? In that case we will need the SSL example in the docs, I think.

**Comment by jamezp** [comment]
@nziakova Yes. I'm going to add an `ssl-context` attribute which can reference an Elytron SSL Context. I'm also adding a test for it, but that's currently where I'm stuck because it isn't working yet :) Hopefully I'll get it done today though.

I'll also update the documentation and the analysis doc. Thanks for catching this! I made some really bad assumptions.

**Comment by nziakova** [comment]
@jamezp It's new to me too, I was just searching for the correct config to test it. Thanks for taking care of it!

**Comment by jamezp** [comment]
Docs have been updated with an SSL example.

**Pull request merged**